staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -217,6 +217,9 @@ func (u *udsGRPCConnector) connect(_ context.Context) (proxier, error) {
 	// See https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/357.
 	tunnelCtx := context.TODO()
 	tunnel, err := client.CreateSingleUseGrpcTunnel(tunnelCtx, udsName, dialOption,
+		grpc.WithBlock(),
```

Comment:
non-blocking comment: WithBlock() is now redundant since WithReturnConnectionError sets it https://github.com/kubernetes/kubernetes/blob/master/vendor/google.golang.org/grpc/dialoptions.go#L284. 

It might good to leave it here though so it's obvious going forward that we're blocking on dial.

Reply:
I think keeping explicit makes sense.